src/v/cluster/partition.cc

```diff
@@ -202,8 +204,24 @@ ss::future<std::error_code> partition::prefix_truncate(
           _raft->ntp());
         co_return make_error_code(cluster::errc::feature_disabled);
     }
-    co_return co_await _log_eviction_stm->truncate(
-      truncation_offset, deadline, _as);
+    vlog(
+      clusterlog.info,
+      "Truncating to redpanda offset {} kafka offset {}",
+      truncation_offset,
+      kafka_truncation_offset);
+    if (truncation_offset == model::offset{}) {
```

Comment:
In the `offset{}` path, it looks like this commit also changes log_eviction_stm to always return errc::success, so why call it at all?